llvm/lib/Target/RISCV/AsmParser/RISCVAsmParser.cpp

```diff
@@ -1651,6 +1651,12 @@ MCRegister RISCVAsmParser::matchRegisterNameHelper(StringRef Name) const {
     Reg = MatchRegisterAltName(Name);
   if (isRVE() && Reg >= RISCV::X16 && Reg <= RISCV::X31)
     Reg = RISCV::NoRegister;
+
+  // Replace vector tuple with the starting register, e.g. V4M4_V8M4 -> V4
+  for (int i = 30; i >= 0; --i)
```

Comment:
What caused us to need this change?

Reply:
It's for inline asm, if we don't have this, `inline-asm.ll` would fail.

Reply:
Does that mean that inline assembly with -S will generate something that contains a register name that binutils can't parse?

Reply:
Yes, it would generate something like
```
error: cannot compile inline asm
<inline asm>:1:13: error: invalid operand for instruction
        vsseg7e8.v V8_V9_V10_V11_V12_V13_V14, (a1)
```

Reply:
Is it possible to change the printing for tuple registers in `RISCVAsmPrinter::PrintAsmOperand` so that it prints just `v8`?

Reply:
I think we can just give vector tuple registers names so that they can be recognized by `AsmParser` correctly, the patch is submitted: https://github.com/llvm/llvm-project/pull/102726